src/ui/ResultList/ResultList.ts

```diff
@@ -800,7 +800,9 @@ export class ResultList extends Component {
     if (!this.options.resultContainer) {
       const elemType = this.options.layout === 'table' ? 'table' : 'div';
       this.options.resultContainer = $$(elemType, { className: 'coveo-result-list-container' }).el;
-      this.element.appendChild(this.options.resultContainer);
+      if (!this.element.classList.contains('CoveoOmniboxResultList')) {
```

Comment:
Thanks for investigating further. While I think we can get away with adding an exception like this, it is not very clean. Looking at the doc of the ResultList:

```
* @param element The HTMLElement on which to instantiate the component.
```

So we are no performing a different behaviour on the passed `element` in part when it has the class `CoveoOmniboxResultList`, hmmm.

Maybe we can leverage the public methods on the ResultList like `disable` on init that use CSS to hide the ResultList? and then `enable` at the right time? This could cause some problems since `disable` stops the component from listening to events, but it may work.

Another option could be to add a constructor argument, `appendResultContainerToElement` that is true by default but false for the OmniboxResultList initialization. That said, it is not clean either.

I feel there might still be a better solution. I would discuss with @olamothe,

At the moment between adding an exception, or your initial approach of emptying the children in the OmniboxResultList component, I think I prefer your initial attempt because it is more encapsulated.

Reply:
Ok I'll talk with him